File: c/gendate.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int is_leap_year(int year) {
    return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0);
}

int main() {
	int is_end, is_leap;
	char iso_date[11];
	time_t current_sec = 0;
	struct tm *current_date = gmtime(&current_sec);

	strftime(iso_date, sizeof(iso_date), "%Y-%m-%d", current_date);
	printf("%s\n", iso_date);

	while (1) {
		current_sec += 86400;
		current_date = gmtime(&current_sec);

		is_leap = (
			!is_leap_year(current_date->tm_year + 1900) &&
			current_date->tm_mon == 1 &&
			current_date->tm_mday == 29
		);

		if (is_leap) {
			continue;
		}

		strftime(iso_date, sizeof(iso_date), "%Y-%m-%d", current_date);
		printf("%s\n", iso_date);

		is_end = (
			(current_date->tm_year == 1100) &&
			(current_date->tm_mon  == 0)    &&
			(current_date->tm_mday == 1)
		);

		if (is_end) {
			break;
		}
	}

	return 0;
}

File: src/py/gendate.py
#!/usr/bin/env python3

from datetime import datetime, timedelta

def is_leap_year(year):
	return (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0)

def main():
	is_end, is_leap = False, False
	current_date = datetime(1970, 1, 1)

	print(current_date.strftime("%Y-%m-%d"))

	while True:
		current_date += timedelta(days=1)

		is_leap = (not is_leap_year(current_date.year) and
				   current_date.month == 2 and
				   current_date.day == 29)

		if is_leap:
			continue

		print(current_date.strftime("%Y-%m-%d"))

		is_end = (current_date.year == 3000 and
				  current_date.month == 1 and
				  current_date.day == 1)

		if is_end:
			break

if __name__ == "__main__":
	main()
